SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialRPCService.h

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+
+#include "Schema/RPCPayload.h"
+#include "Utils/RPCRingBuffer.h"
+
+#include <WorkerSDK/improbable/c_schema.h>
+#include <WorkerSDK/improbable/c_worker.h>
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialRPCService, Log, All);
+
+class USpatialStaticComponentView;
+struct RPCRingBuffer;
+
+DECLARE_DELEGATE_RetVal_ThreeParams(bool, ExtractRPCDelegate, Worker_EntityId, ERPCType, const SpatialGDK::RPCPayload&);
+
+namespace SpatialGDK
+{
+
+struct EntityRPCType
+{
+	EntityRPCType(Worker_EntityId EntityId, ERPCType Type)
+		: EntityId(EntityId)
+		, Type(Type)
+	{}
+
+	Worker_EntityId EntityId;
+	ERPCType Type;
+
+	inline bool operator==(const EntityRPCType& Other) const
+	{
+		return EntityId == Other.EntityId && Type == Other.Type;
+	}
+
+	friend inline uint32 GetTypeHash(EntityRPCType Value)
```

Comment:
nit - mildly prefer not having `inline` on functions defined in the class definition, but it doesn't really hurt things.